server/src/main/java/io/crate/expression/scalar/CurrentDateFunction.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.expression.scalar;
+
+import io.crate.data.Input;
+import io.crate.metadata.NodeContext;
+import io.crate.metadata.Scalar;
+import io.crate.metadata.TransactionContext;
+import io.crate.metadata.functions.Signature;
+import io.crate.types.DataTypes;
+
+import java.time.LocalDate;
+import java.time.ZoneOffset;
+
+public final class CurrentDateFunction extends Scalar<LocalDate, String> {
+
+    public static final String NAME = "curdate";
+
+    public static void register(ScalarFunctionModule module) {
+        module.register(
+            Signature.scalar(
+                NAME,
+                DataTypes.DATE.getTypeSignature()
+            ),
+            CurrentDateFunction::new
+        );
+    }
+
+    private final Signature signature;
+    private final Signature boundSignature;
+
+    public CurrentDateFunction(Signature signature, Signature boundSignature) {
+        this.signature = signature;
+        this.boundSignature = boundSignature;
+    }
+
+    @Override
+    @SafeVarargs
+    public final LocalDate evaluate(TransactionContext txnCtx, NodeContext nodeCtx, Input<String>... args) {
```

Comment:
We've just saw that this seems odd as the returned data type here must be a CrateDB data type and must align with the signatures. So I should be `Date` instead. Isn't `current_date` currently failing because of that when used in full integration test or manual test by e.g. using `crash`?

Reply:
Good catch, thanks.
<details>

<summary> current_date works locally (screenshot under the spoiler) </summary>

![Capture](https://user-images.githubusercontent.com/7446940/109770117-6ff0ec80-7bfb-11eb-943a-08a31ed613b0.PNG)

</details>

However, `current_date` calls `DateType.implicitCast` which tries to cast `LocalDate` to `Long` causing Exception in `Scalar.normalizeSymbol`. It works because exception case is covered but that casting issue must be avoided indeed. 

I have tried locally 3 approaches:

1. Keep current implementation but add "instance of LocalDate' check to avoid casting exception
2. 
pros 

- no need to create wrapper class
- simple Long type is used as internal type 

cons

- no control over output format - basically it's LocalDate's default toString which uses `ISO-8601`
- as you mentioned, signature mismatch looks odd

2. 
Use wrapper class (not LocalDate but own lightweight class, just long millis holder) - idea similar to TimeTZ class
pros
- control over format - can specify any format in `toString` and later improve `curdate()` to accept date format.
- signature is still mismatching with `module.register` but I guess in this in case it's better - as evaluate return type matches with `extends DataType<WrapperClass>`

 cons
 
 - <details>
 <summary>curdate() returns 2 rows as a result: `toString() + timestamp(?)` - haven't figured out how to avoid this.</summary>
 
 
![CaptureUtcDate](https://user-images.githubusercontent.com/7446940/109825565-3be8ec00-7c3a-11eb-9f3b-4d4e0a58aa31.PNG)

 </details>

3. Do not create wrapper class but just return DateType in evaluate. Add long millis as a fiield and add toString with needed format.

pros 
- signature matches and no need to create wrapper class
cons  
- not consistent with other functions - I haven't seen any function returning not internal type but type itself, even current_time uses not TimeTZType but TimeTZ.


Let me know which option is better

Reply:
I implemented second option from my comment above. 

<details>

<summary>
I tested it locally, it returns date and also timestamp - 
</summary>

![Capture_all_3_with_second_row](https://user-images.githubusercontent.com/7446940/110487973-5fa2ab00-80ee-11eb-86da-bf5e73083daa.PNG)

</details

As showed in the screenshot, all time/date related functions return 2 rows  - I couldn't get rid of timestamp but as `current_time` and `current_timestamp` also return 2 rows I assumed that it's expected behavior.

Reply:
@BaurzhanSakhariev 

Sorry for the late reply and it also seems that I was also confused by the `current_time` implementation and pushed you maybe in the wrong direction.

`current_time` will respond an `array(long)` where the 1st item is the current time since epoch and the 2nd the timezone offset. That is needed by clients to render the current time correctly.

But for `current_date` no timezone information is needed and exists.

So we can just return a `Long` on the evaluate method and no wrapper or `rest` specific json formatting is needed. (which we did for `TimeTZ`, see https://github.com/crate/crate/blob/master/server/src/main/java/io/crate/types/DataTypeXContentExtension.java#L67).

Reply:
Making it human readable is up to the client, so `crash` should do that which it currently doesn't (we should create an issue/fix there). Otherwise the client would have to parse a human readable format even if it would never be presented to humans.

Reply:
Ah, I see, I misunderstood the task. 
I saw select example in the [issue ](https://github.com/crate/crate/issues/9651) and had a wrong impression that function itself should deliver ready-to-present result. That clarification makes things simpler, thanks!

Reply:
> I saw select example in the issue and had a wrong impression that function itself should deliver ready-to-present result.

Ah I see. Sorry for all this confusion!